sui_core/src/authority/authority_store.rs

```diff
@@ -770,11 +718,32 @@ impl<const ALL_OBJ_VER: bool, S: Eq + Serialize + for<'de> Deserialize<'de>>
                 )?;
             }
 
-            // Atomic write of all locks & other data
+            // Atomic write of all data other than locks
             write_batch.write()?;
 
+            // Initialize object locks for new objects.  After this point, transactions on new objects
+            // can run.  So it is critical this is done AFTER objects are done writing.
+            // TODO: what if we fail to initialize the locks?  That should NOT happen, but try rolling back
```

Comment:
Can we file an issue for this? It seems that this may be a challenging problem to solve?

Reply:
> So it is critical this is done AFTER objects are done writing.

The DB has a [`notify_read` operation](https://github.com/MystenLabs/mysten-infra/blob/main/crates/typed-store/src/lib.rs#L215-L227) which does not necessitate that a key exists to launch, and since rocksdb will read its writes, you can use it to check write success.

Reply:
Issue filed.